### PR TITLE
refactor(CNV-51887): move libdv out of functional tests

### PR DESCRIPTION
**Pull request by codingben**
### What this PR does

The libdv package contains NewDataVolume
function options that helps us to construct
a new v1.VirtualMachine.

Currently libdv used only in functional tests
and not importable by other projects, because
tests likely tend to be changed or removed.

Move libdv package to pkg folder that is likely
not to be changed like libvmi package.

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/clone_test.go [hunk 1]
```diff
@@ -13,8 +13,8 @@ import (
 	"kubevirt.io/kubevirt/pkg/libvmi"
 	"kubevirt.io/kubevirt/pkg/pointer"
```
Comment (0xFelix) [inline tests/clone_test.go:15]: nit: Move to block above

**Review by 0xFelix — COMMENTED** [review]
Important question: Who is going to own libdv? Who is going to maintain it?

If we should agree to move `libdv` to `pkg`, it needs an `OWNER` file and a `README.md`, similar to `libvmi`.

**Review by akalenyu — COMMENTED** [review]
> Important question: Who is going to own libdv? Who is going to maintain it?
> 
> If we should agree to move `libdv` to `pkg`, it needs an `OWNER` file and a `README.md`, similar to `libvmi`.

sig-storage could own it, but my worry is that kubevirt/kubevirt as it is today is not really importable.

@xpivarc on one of kubevirt-dev's threads:
>  I would generally discourage usage of any code from Kubevirt. We do not provide any compatibility guruanties (exception being API definitions), even for client-go. We recommend generating the client based on API.

So I am not sure how straightforward it would be to import something from kubevirt/kubevirt.
Maybe you could try importing libvmi and see if that works, before committing to libdv?

**Comment by codingben** [comment]
@0xFelix I see that Alex replied to your question about ownership of this pkg.

> Maybe you could try importing libvmi and see if that works, before committing to libdv?

@akalenyu We just did an import of libvmi, and added an additional new function ([this one](https://github.com/kubevirt/kubevirt-tekton-tasks/blob/5df1dedc12a65b33027e99b87cf48677692f329a/test/disk_uploader_test.go#L121-L144)) instead of using libdv's options.

**Comment by akalenyu** [comment]
> @0xFelix I see that Alex replied to your question about ownership of this pkg.
> 
> > Maybe you could try importing libvmi and see if that works, before committing to libdv?
> 
> @akalenyu We just did an import of libvmi, and added an additional new function ([this one](https://github.com/kubevirt/kubevirt-tekton-tasks/blob/5df1dedc12a65b33027e99b87cf48677692f329a/test/disk_uploader_test.go#L121-L144)) instead of using libdv's options.

Ok sounds good, in that case, could you add the owners file so sig-storage could "properly" own it?

**Comment by codingben** [comment]
> > @0xFelix I see that Alex replied to your question about ownership of this pkg.
> > > Maybe you could try importing libvmi and see if that works, before committing to libdv?
> > 
> > 
> > @akalenyu We just did an import of libvmi, and added an additional new function ([this one](https://github.com/kubevirt/kubevirt-tekton-tasks/blob/5df1dedc12a65b33027e99b87cf48677692f329a/test/disk_uploader_test.go#L121-L144)) instead of using libdv's options.
> 
> Ok sounds good, in that case, could you add the owners file so sig-storage could "properly" own it?

Thanks. Yes, I've added owners file.

File: tests/clone_test.go (hunk 1 above)
Comment (codingben) [reply tests/clone_test.go:16]: Done in this [revision](https://github.com/kubevirt/kubevirt/compare/eed4dfd64c3d0d8a8b9fff48933f75ff830f8959..3aa64ad2b9e147718c1376d3bede0ee7eb643671).

**Comment by 0xFelix** [comment]
I'm still undecided about this. Can we take it to `kubevirt-dev` first to see what others think?

**Comment by codingben** [comment]
> I'm still undecided about this. Can we take it to `kubevirt-dev` first to see what others think?

What do you think we can achieve from `kubevirt-dev`? We have @akalenyu's approval from storage team to be the owner of this package. Do you have concerns by placing this package inside kubevirt/pkg at all, or something else?

**Comment by 0xFelix** [comment]
@akalenyu mentioned this:

> ... my worry is that kubevirt/kubevirt as it is today is not really importable.

Although it works in your case, it is still an issue. I'm not sure if we should promote importing `kubevirt/kubevirt`.

**Comment by codingben** [comment]
> @akalenyu mentioned this:
> 
> > ... my worry is that kubevirt/kubevirt as it is today is not really importable.
> 
> Although it works in your case, it is still an issue. I'm not sure if we should promote importing `kubevirt/kubevirt`.

Ack.

**Comment by codingben** [comment]
> I'm still undecided about this. Can we take it to `kubevirt-dev` first to see what others think?

Done, @0xFelix @akalenyu please add your opinion here: https://groups.google.com/g/kubevirt-dev/c/FBOrghostS4

File: pkg/libdv/dv.go [hunk 2]
```diff
@@ -24,16 +24,16 @@ import (
...
 	instancetypeapi "kubevirt.io/api/instancetype"
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 
 	"kubevirt.io/kubevirt/pkg/pointer"
-	cd "kubevirt.io/kubevirt/tests/containerdisk"
 )
 
 const (
 	dvRandomNameLength = 12
+	cirrosVolumeSize   = "512Mi"
```
[the remark this answers is not in the document]
Comment (codingben) [reply pkg/libdv/dv.go:36]: I wonder if we should name it as `cirrosVolumeSize` and not `defaultVolumeSize`. What do you think? `StorageWithVolumeSize` can be used to override the volume size if needed.

[the remark this answers is not in the document]
Comment (akalenyu) [reply pkg/libdv/dv.go:36]: defaultVolumeSize sounds good

[the remark this answers is not in the document]
Comment (codingben) [reply pkg/libdv/dv.go]: Done, can you please review it again? :pray:

File: pkg/libdv/dv.go [hunk 3]
```diff
@@ -24,16 +24,16 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/rand"
-	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 
 	instancetypeapi "kubevirt.io/api/instancetype"
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 
 	"kubevirt.io/kubevirt/pkg/pointer"
-	cd "kubevirt.io/kubevirt/tests/containerdisk"
```
Comment (iholder101) [inline pkg/libdv/dv.go:32]: That's great!

**Review by iholder101 — COMMENTED** [review]
> > I'm still undecided about this. Can we take it to `kubevirt-dev` first to see what others think?
> 
> Done, @0xFelix @akalenyu please add your opinion here: [groups.google.com/g/kubevirt-dev/c/FBOrghostS4](https://groups.google.com/g/kubevirt-dev/c/FBOrghostS4)

Following the discussion in the mailing list, in which multiple people were in favor of this, and due to the fact that I couldn't find a compelling argument against this, I'll approve the PR.

Thank you @codingben for the PR and @0xFelix for the discussion!

We indeed still need to ensure packages like this (and libvmi) are externally importable, which I'd be happy to see in follow-ups. Either way, after this PR these packages are at least importable by production code and are safe from test-specific code slipping in.

/approve

**Review by akalenyu — COMMENTED** [review]
/lgtm
thanks!

**Pull request merged**